tests/Tests/Regressions.hs

```diff
@@ -105,8 +109,8 @@ t227 =
 t301 :: IO ()
 t301 = do
     assertEqual "The length of the array remains the same despite slicing"
-                (TA.length originalArr)
-                (TA.length newArr)
+                (I# (sizeofByteArray# (TA.aBA originalArr)))
+                (I# (sizeofByteArray# (TA.aBA newArr)))
```

Comment:
Good question!

`TA.length` is defined only when the package is built in `developer` mode:
https://github.com/haskell/text/blob/6d8ae4d5e7df8c343d1b9bbcc1cd64db2f79ff19/src/Data/Text/Array.hs#L35-L37

The existing test suite lists `Data.Test.Array` as its own module and thus can enforce `-DASSERTS` irrespective to actual `developer` flag:
https://github.com/haskell/text/blob/6d8ae4d5e7df8c343d1b9bbcc1cd64db2f79ff19/tests/text-tests.cabal#L89-L90

Since now the test suite depends on a proper `text` package, it lost its ability to enforce `-DASSERTS` on its modules, and now this is guided by `developer` flag only. In order to compile tests when this flag is not enabled, I had to avoid `TA.length` and replace it with `sizeOfByteArray#`. 

It would be nice to put `constraint: text +developer` into `cabal.project`, so that all developers (and CI) have it enabled by default. Unfortunately, such configuration breaks benchmarks. As mentioned at the bottom of https://github.com/haskell/text/pull/314#issue-588849330, I'll fix this in an upcoming PR.